pkg/auth/oauth2authenticator.go

```diff
@@ -0,0 +1,287 @@
+package auth
+
+import (
+	"context"
+	crypto_rand "crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"html"
+	"math/rand"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/pkg/browser"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"golang.org/x/oauth2"
+)
+
+const (
+	CONFIG_KEY_OAUTH_TOKEN string        = "OAUTH_TOKEN"
+	OAUTH_CLIENT_ID        string        = "0oa37b7oa3zOoDWCe4h7"
+	CALLBACK_HOSTNAME      string        = "127.0.0.1"
+	CALLBACK_PATH          string        = "/authorization-code/callback"
+	TIMEOUT_SECONDS        time.Duration = 120 * time.Second
+	SnykOAuthTokenEnvKey   string        = "SNYK_OAUTH_TOKEN"
+)
+
+var _ Authenticator = (*oAuth2Authenticator)(nil)
+
+var acceptedCallbackPorts = []int{8080, 18081, 28082, 38083, 48084}
+
+type oAuth2Authenticator struct {
+	httpClient         *http.Client
+	config             configuration.Configuration
+	oauthConfig        *oauth2.Config
+	token              *oauth2.Token
+	headless           bool
+	openBrowserFunc    func(authUrl string)
+	shutdownServerFunc func(server *http.Server)
+}
+
+func (o *oAuth2Authenticator) AddEnvironmentVariables(env []string) ([]string, error) {
```

Comment:
We may want to refactor this in a later PR - I don't think this should be here. Let's add a todo and move on for now.